frontend/src/views/Room.jsx

```diff
@@ -1,19 +1,107 @@
+  // we eventually need the full list of participants, not only the visible ones
+  // this array will hold data such as name foor the purpose of
+  // having a participants name list, etc.
+
+  // It's possible that visibleParitcipants will eventually not need to be in the state,
+  // if it ends up being a component property passed in by the parent component.
+  // for now it needs to be in the state so that the "Add Participant" test button can work
+  const [visibleParticipants, setVisibleParticipants] = useState([]);
+  const rowsLimit = { xs: 2, sm: 2, md: 2 };
+  const tilesPerRowLimit = { xs: 2, sm: 6, md: 10 };
+  // it's assumed that we'll get a maximum of rowsLimit*tilesPerRowLimit visibleParticipants
+  // if this precondition isn't true then the video grid can't be expected to render properly
+  const calculateTilesPerRow = (screenSize) => {
+    // screenSize should be either 'xs', 'sm' or 'md'
+    const tilesAmount = visibleParticipants.length + 1;
+    // the "+ 1" is because of the user tile
+    return Math.min(
+      tilesPerRowLimit[screenSize],
+      Math.ceil(tilesAmount / rowsLimit[screenSize]),
+    );
+  };
+  const tilesPerRow = {
+    xs: calculateTilesPerRow('xs'),
+    sm: calculateTilesPerRow('sm'),
+    md: calculateTilesPerRow('md'),
+  };
   const roomId = useLoaderData();
+  const addParticipant = () => {
+    // adds a dummy participant to test the view's adaptiveness
+    // to the amount of participants in the room
+    const participantsCopy = [...visibleParticipants, { id: idCount }];
+    setIdCount(idCount + 1);
+    setVisibleParticipants(participantsCopy);
+  };
+  // for testing purposes, we use the user camera stream in every video component
+  const [userStream, setUserStream] = useState(null);
+  useEffect(() => {
+    const getUserCameraStream = async () => {
+      try {
+        const stream = await navigator.mediaDevices.getUserMedia({ audio: false, video: true });
+        setUserStream(stream);
+      } catch (err) {
+        console.log('error trying to get user camera stream');
+      }
+    };
+    getUserCameraStream();
+  }, []);
   return (
-    <div>
-      This will eventually be a room with ID:
-      {' '}
-      {roomId}
-    </div>
+    <>
+      <Typography variant="h4" component="h1">
+        Room
+        {' '}
+        {roomId}
+      </Typography>
+      {userStream && (
+      <Grid container spacing={2} columns={tilesPerRow} alignItems="center" justifyContent="center">
+        {visibleParticipants.map((participant) => (
+          <Grid item xs={1} sm={1} md={1} key={participant.id}>
+            <div style={{ borderStyle: 'solid' }}>
+              <Video
+                stream={userStream}
+              />
+            </div>
+          </Grid>
+        ))}
+        <Grid item xs={1} sm={1} md={1}>
+          <div style={{ borderStyle: 'solid' }}>
+            <Video
+              stream={userStream}
+            />
+          </div>
+        </Grid>
```

Comment:
Is this the tile for the 'self' video? Personally I'd prefer keeping it as an absolute-positioned element in the bottom right (for example) rather than as part of the tiles grid, but I'm interested in hearing what people think.

Also, let's consider we'll have different view layouts (grid and presenter at least, I believe), so we might just adopt different strategies in each.

Reply:
I agree with having it bottom right, fixed position always

Reply:
Yeap I also agree. By placing it at the end of the gird, because of how the grid gets rendered, it will always get placed at the bottom right of the grid. Is that sufficient for now or do we need to take it out of the grid for some other reason?

Reply:
The only way it wouldn't end up in the bottom right is if the precondition of `visibleParticipants` <= `rowsLimit*tilesPerRowLimit` fails to be met

Reply:
Nevermind I think I've got what you guys are saying, completely out of the grid, fixed in the bottom right of the view. Got it

Reply:
Right. What I mean by fixed position is out of the grid, but that's a matter of personal preference. We could always get back to it later if we want :)

Personally the downsides I see with having your own video as part of the grid are:
- we "lose" one slot for a potential remote stream that we need to hide, so that the own video is always visible
- the position of the local video in the page will move as people join and leave, so you sort of have to look for it if you want to see yourself (minor issue, but adds a little friction)
- you don't get a clear separation of local streams vs remote streams. Potentially there are cases in which you might be sharing more than one stream (video and screenshare, for example), so a clearer separation helps

But as I said, these are not major issues, so we can just as well keep the grid as it is, if it means we can move on faster with the project ;)

Reply:
Yes I see now, I agree we should take it out of the grid for sure